Update identifier reference (#1655)
open-contracting/standard: docs/schema/identifiers.md

* Organization identifiers are important to know who is involved in each contract;
* Release, award and contract identifiers are important to help cross-reference information.

## Open contracting process identifier (ocid)

<img src="../../_static/svg/green_compilation.svg" width="150" align="right"/>

OCDS defines a release's `ocid` field as:

```{field-description} ../../build/current_lang/release-schema.json /properties/ocid
```

(The definition of a record's `ocid` field replaces "release" with "record".)

Since the `ocid` field is a required field and is globally unique, it can be used to join up data that is published at different times or from different systems.

An `ocid` is composed of two parts:

1. An [OCID prefix](#ocid-prefix), which identifies the series of identifiers to which the `ocid` belongs
1. An internal identifier for the contracting (or planning) process, which is unique within the series

It is encouraged to separate the OCID prefix and the internal identifier with a hyphen (`-`).

The `ocid` is case-sensitive; in other words, the letter case of an ocid must be consistent.

> To assign an `ocid` to a contracting (or planning) process, you need to register an OCID prefix and choose an internal identifier.

Reviewer: Add a link to the guidance on registering an OCID prefix to "register an OCID prefix", or to the section later in the page "OCID prefix"

Author: Done in https://github.com/open-contracting/standard/pull/1655/commits/eda23e89bd4de753a8d977ca767dd3dcde40e591

> ````{admonition} Worked Example

Reviewer 2: Change in all cases to:

```suggestion
````{admonition} Example
```

Author: Done in 36911186b9da46fa4a6aec246405596ae6b966ff

:class: hint

Two publishers, the UK Atomic Energy Authority and Health Canada, use sequential numbers as internal identifiers for their contracting processes.

> The UK Atomic Energy Authority initiates a contracting process to purchase productivity software and assigns it the internal identifier '0005'. Health Canada initiates a contracting process to purchase office furniture and also assigns it the internal identifier '0005'.

Reviewer 2: Use double quotes instead of single quotes in all examples. Only use single quotes for codes.

Author: Done in 36911186b9da46fa4a6aec246405596ae6b966ff

> To create a globally unique `ocid`, each publisher prepends their internal identifier with their OCID prefix: 'ocds-fh349f' for the UK Atomic Energy Authority and 'ocds-twb234 for Health Canada.

Reviewer 2: Missing a closing quote on the second prefix.

Also, these OCID prefixes don’t actually exist. Maybe use real publishers, with corresponding prefixes?

Author: Missing quote fixed in 36911186b9da46fa4a6aec246405596ae6b966ff. I also corrected instances of smart quotation marks.

There's some background on the choice of publishers in https://github.com/open-contracting/standard/issues/1595#issuecomment-1752226485, in addition to which I thought that using organisations with the names of their countries in their names would clearly communicate that the publications are distinct.

Do you think that using real prefixes combined with fictional internal identifiers is preferable to using fictional prefixes with fictional identifiers? I figured that combining real and fictional might be confusing for readers familiar with the real publisher.

If you prefer to use a real prefix with a fictional internal identifier, we could replace the UK publisher with Find a Tender Service and the Canada publisher with Buy and Sell.

Reviewer 2: Hmm, maybe just change the prefixes to def456 and ghi789?

Author: Sounds good. Done in 3931a15ef839e42b5398a67c3bf4eaf7a4b26d8a

The UK Atomic Energy Authority assigns the `ocid` 'ocds-fh349f-0005'.

```json
{
  "ocid": "ocds-fh349f-0005",
  "publisher": {
    "name": "UK Atomic Energy Authority"
  },
  "tender": {
    "id": "0005",
    "title": "Productivity software"
  }
}
```

Health Canada assigns the `ocid` 'ocds-twb234-0005'.

```json
{
  "ocid": "ocds-twb234-0005",
  "publisher": {
    "name": "Health Canada"
  },
  "tender": {
    "id": "0005",
    "title": "Office furniture"
  }
}
```

> As a result, users and tools that work with data from both publishers are less likely to confuse the two contracting processes. If another publisher, such as an independent monitor, wants to publish more data about one of the contracting processes, they can reference it using its unique `ocid`.

Reviewer 2: Removed content that overlaps with #364

```suggestion
As such, users and tools that work with both publishers' data will not confuse the two contracting processes.
```

The original content was:

> And, if an independent civil society contract monitoring group want to publish a report about implementation of Town A's road project, or Town B's text-book procurement, they have distinct identifiers they can use in their own data to refer to these.

But this is about references from a report or dataset, not from an OCDS dataset specifically.

* A prefix for the organization list (also known as registry or register) from which the identifier is drawn: for example, a company register.
* The existing identifier for the organization within that list.

In OCDS, the list's prefix is disclosed via the `scheme` field of an identifier object, with the existing identifier in the `id` field. If there is a recognized public URL that uniquely identifies the organization, this can be disclosed via the `uri` field.

````{admonition} Worked Example
:class: hint

> Companies in the UK are assigned identifiers by [Companies House](https://www.gov.uk/government/organisations/companies-house). MICROSOFT LIMITED's identifier is '01624297' and the prefix for Companies House is "GB-COH". Therefore, MICROSOFT LIMITED's organization identifier can be expressed as follows:

Reviewer 2: ```suggestion
[Companies House](https://www.gov.uk/government/organisations/companies-house) registers company information in the United Kingdom. The prefix for Companies House is "GB-COH”. Companies House registered a company with the name "MICROSOFT LIMITED" and the number “01624297”. The company's organization identifier can be expressed as:
```

```json
{
  "parties": [
    {
      "identifier": {
        "scheme": "GB-COH",
        "id": "01624297",
        "legalName": "MICROSOFT LIMITED",

>         "uri": "http://data.companieshouse.gov.uk/doc/company/01624297"

Reviewer 2: ```suggestion
        "uri": "https://data.companieshouse.gov.uk/doc/company/01624297"
```
